delivery_carrier_label_postlogistics/models/delivery.py

```diff
@@ -105,29 +105,25 @@ class DeliveryCarrierOption(models.Model):
 
     name = fields.Char(translate=True)
 
-    def fields_view_get(self, cr, uid, view_id=None, view_type='form',
-                        context=None, toolbar=False, submenu=False):
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form',
+                        toolbar=False, submenu=False):
         _super = super(DeliveryCarrierOption, self)
-        result = _super.fields_view_get(cr, uid, view_id=view_id,
-                                        view_type=view_type, context=context,
+        result = _super.fields_view_get(view_id=view_id,
+                                        view_type=view_type,
                                         toolbar=toolbar, submenu=submenu)
-        xmlid = 'delivery_carrier_label_postlogistics.postlogistics'
-        ref = self.pool['ir.model.data'].xmlid_to_object
-        postlogistics_partner = ref(cr, uid, xmlid, context=context)
-        if context.get('default_carrier_id'):
-            carrier_obj = self.pool['delivery.carrier']
-            carrier = carrier_obj.browse(cr, uid,
-                                         context['default_carrier_id'],
-                                         context=context)
-            if carrier.partner_id == postlogistics_partner:
+        default_carrier_id = self.env.context.get('default_carrier_id')
+        if default_carrier_id:
+            carrier = self.env['delivery.carrier'].browse(default_carrier_id)
+            if carrier.is_postlogistics():
                 arch = result['arch']
                 doc = etree.fromstring(arch)
                 for node in doc.xpath("//field[@name='tmpl_option_id']"):
                     node.set(
                         'domain',
-                        "[('partner_id', '=', %s), "
+                        "[('product_id', '=', %s), "
```

Comment:
There is no `product_id` on `delivery.carrier.template.option` model, what brings :
```
ValueError: Invalid field u'product_id' in leaf "<osv.ExtendedLeaf: (u'product_id', u'=', 2) on delivery_carrier_template_option (ctx: )>"
```